llvm/lib/Target/DirectX/DXILDataScalarization.cpp

```diff
@@ -27,6 +27,20 @@ static const int MaxVecSize = 4;
 
 using namespace llvm;
 
+// Recursively creates an array-like version of a given vector type.
+static Type *equivalentArrayTypeFromVector(Type *T) {
```

Comment:
This is fine I would have just made a function declaration at the top so the implementation could live anywhere.